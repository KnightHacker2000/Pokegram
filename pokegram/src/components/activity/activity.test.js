/**
 * @jest-environment jsdom
 */
import React from 'react';
import { expect, test } from '@jest/globals';
import renderer from 'react-test-renderer';
import Act from './activity';
import '@testing-library/jest-dom/extend-expect';
import Activity from '../../models/activity';

test('test activity snapshot', () => {
  const actComp = renderer.create(<Act.Act />);
  const tree = actComp.toJSON();
  // console.log(loginComp.toJSON());
  expect(tree).toMatchSnapshot();
});

describe("Parse Text-1", () => {
  const act = new Activity();
  act.id = 1;
  act.initiatorId = 2;
  act.targetId = 1;
  act.activityType = 'Follow';
  act.timestamp = Date.now;

  test("check parseActText ", () => {
    expect(Act.parseActText(act)).toBe(' followed you!');
  });
});

describe("Parse Text-2", () => {
  const act = new Activity();
  act.id = 1;
  act.initiatorId = 2;
  act.targetId = 1;
  act.activityType = 'Unfollow';
  act.timestamp = Date.now;

  test("check parseActText ", () => {
    expect(Act.parseActText(act)).toBe(' unfollowed you!');
  });
});

describe("Parse Text-3", () => {
  const act = new Activity();
  act.id = 1;
  act.initiatorId = 2;
  act.targetId = 1;
  act.activityType = 'Like';
  act.timestamp = Date.now;

  test("check parseActText ", () => {
    expect(Act.parseActText(act)).toBe(' liked your post!');
  });
});

describe("Parse Text-4", () => {
  const act = new Activity();
  act.id = 1;
  act.initiatorId = 2;
  act.targetId = 1;
  act.activityType = 'Comment';
  act.timestamp = Date.now;

  test("check parseActText ", () => {
    expect(Act.parseActText(act)).toBe(' commented your post!');
  });
});