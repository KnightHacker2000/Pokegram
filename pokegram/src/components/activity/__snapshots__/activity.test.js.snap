// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test activity snapshot 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "paddingTop": "10px",
    }
  }
>
  <ul
    className="MuiList-root MuiList-padding css-1mcnwpj-MuiList-root"
  />
</div>
`;
